src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  removeTodoFromServer: (id: number) => void;
+  updateTodoOnServer: (todo: Todo) => void;
+  updatingStage: number[];
+  handleEditingTodo: (id: number) => void;
+  editedTodoId: number;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  removeTodoFromServer,
+  updateTodoOnServer,
+  updatingStage,
+  handleEditingTodo,
+  editedTodoId,
+}) => {
+  const [tempTitle, setTempTitle] = useState(todo.title);
+
+  const { title, completed, id } = todo;
+  const isBeingEditedNow = editedTodoId === id;
+
+  const handleUpdateStatus = (prevTodo: Todo) => {
+    updateTodoOnServer({
+      ...prevTodo,
+      completed: !completed,
+    });
+  };
+
+  const handleUpdateTitle = (prevTodo: Todo) => {
+    updateTodoOnServer({
+      ...prevTodo,
+      title: tempTitle,
+    });
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTempTitle(event.target.value);
+  };
+
+  const saveChanges = () => {
+    handleEditingTodo(0);
+    handleUpdateTitle(todo);
+  };
+
+  const handleBlur = () => {
+    saveChanges();
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    saveChanges();
+  };
+
+  const handleEscape = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setTempTitle(title);
+      handleEditingTodo(0);
+    }
+  };
```

Comment:
I suppose it's better use addEventListener & useEffect here, and maybe change type of event  `React.KeyboardEvent`